enhancements/node-pull-credentials/pull-credentials.md

```diff
@@ -0,0 +1,193 @@
+
+In order to provide a seamless developer experience OpenShift should allow
+users to import or to use during builds any registry credentials provided
+during or after the installation, among the registries we would like to allow
+users to interact with is `registry.redhat.io`. 
+
+Nowadays pull credentials provided during cluster install are available on
+node's filesystem. If user attempts to import an image stream from these
+locations OpenShift fails as neither `openshift-api` nor `builder` use the
+credentials provided during the installation.
+
+### Goals
+
+- Allow users to import image streams from images hosted in any registry
+  provided during or after cluster installation, including `registry.redhat.io`.
+- Allow users to run `builds` based on images hosted in any registry provided
+  during or after cluster installation without providing any extra credentials.
+
+### Non-Goals
+
+- Change the way users manage cluster pull credentials.
+  * Additional credentials may still be provided through secrets on namespace.
+- Use Node mirroring config.
+  * To use config files files such as `/etc/containers/registries.conf` is not
+  part of this proposal.
+- Use Node CAs.
+  * This should be done in the future through another enhancement proposal as
+  we currently have different sources of truth when it comes to CAs(different
+  methods of providing them to different parts of the codebase).
+
+## Proposal
+
+### User Stories
+
+As an OpenShift user
+I want to be able to import imagestreams from any image registry that the
+cluster nodes can pull images from
+So that I can use them without manually creating any extra credentials
+
+As an OpenShift user
+I want to be able to use, during builds, images from any image registry that
+the cluster nodes can pull images from
+So that I can use these images as base for my own images
+
+### Implementation Details/Notes/Constraints
+
+This is the node's filesystem path we are taking into account:
+
+- `/var/lib/kubelet/config.json` contains node's pull credentials.
+
+#### Image Stream Import
+
+- Mount the location as `hostPath` during `openshift-apiserver` deploy under
+  `/node/var/lib/kubelet/config.json`.
+
+```
+volumeMounts:
+- mountPath: /node/var/lib/kubelet/config.json
+  name: node-pull-credentials
+volumes:
+- name: node-pull-credentials
+  hostPath:
+    path: /var/lib/kubelet/config.json
+    type: File
+```
+
+- When user requires to import an image stream, `openshift-apiserver` would
+  merge all pull credentials found in `/node/var/lib/kubelet/config.json` with
+  other credentials that may exist on the namespace.
+
+#### Builds
+
+- `controller-manager` would mount pull credentials inside `build` pod as 
+  `hostPath`(similar to what is done for Image Stream Import).
+- Builder image parses node's pull credentials and uses them during build,
+  merging with other pull credentials that may exist on the same namespace.
```

Comment:
Adding a clarifying note about pull secrets in the `BuildConfig`:

```
- Builder image parses node's pull credentials and uses them during build,
  merging with other pull credentials that are linked to the `builder` service account.
  If the `BuildConfig` specifies a pull secret, we will continue the current behavior of
  using the provided pull secret as an override.
```

Reply:
done.